File: algorithms/searching/binary_search/javascript/binarySearch.test.js
const { binarySearch } = require('./binarySearch');

const array = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10];

it.each([
  { array: array, target: 1, expectation: 0 },
  { array: array, target: 2, expectation: 1 },
  { array: array, target: 3, expectation: 2 },
  { array: array, target: 10, expectation: 9 },
  { array: array, target: 11, expectation: null },
])('finds the target\'s in the given array: %j', (t) => {
  const result = binarySearch(t.array, t.target);
  expect(result).toEqual(t.expectation);
});

File: algorithms/searching/binary_search/ruby/binary_search_test.rb
require_relative "./binary_search"

describe "#binary_search" do
  it "returns the index when the target is in the array" do
    array = (1..24000).to_a
    target = 15_768

    result = binary_search(array, target)

    expect(result).to eq(15_767)
  end

  it "returns the index when the target is in the array" do
    array = (0..250).to_a
    target = 3

    result = binary_search(array, target)

    expect(result).to eq(3)
  end

  it "returns nil when the target is not in the array" do
    array = (0..250).to_a
    target = 312

    result = binary_search(array, target)

    expect(result).to be_nil
  end
end
